lib/pbench/client/__init__.py

```diff
@@ -321,31 +328,27 @@ def connect(self, headers: Optional[dict[str, str]] = None) -> None:
         self.endpoints = response.json()
         assert self.endpoints
 
-    def login(self, user: str, password: str) -> JSONMap:
-        """Login to a specified username with the password, and store the
-        resulting authentication token.
+        # Create an OIDCAdmin object and confirm the connection was successful
+        self.oidc_admin = OIDCAdmin(server_url=self.endpoints["openid"]["server"])
+        assert self.oidc_admin.get_admin_token()["access_token"]
+
+    def login(self, user: str, password: str):
+        """Login to a specified username with the password on OIDC server,
+        and store the resulting authentication token.
 
         Args:
             user:       Account username
             password:   Account password
-
-        Returns:
-            The login response
-        """
-        response = self.post(API.LOGIN, json={"username": user, "password": password})
-        response.raise_for_status()
-        json = response.json()
-        self.username = json["username"]
-        self.auth_token = json["auth_token"]
-        return JSONMap(json)
-
-    def logout(self) -> None:
-        """Logout the currently authenticated user and remove the
-        authentication token.
         """
-        self.post(API.LOGOUT)
-        self.username = None
-        self.auth_token = None
+        response = self.oidc_admin.user_login(
+            client_id=self.endpoints["openid"]["client"],
+            username=user,
+            password=password,
+        )
+        assert response["access_token"]
+        auth_token = response["access_token"]
+        self.username = user
+        self.auth_token = auth_token
```

Comment:
I don't think the local variable `auth_token` is buying us much, here.  Now, if you want to use it in an `if` statement in lieu of the `assert`, that's good; otherwise, you should remove it and just in-line the expression at line 351.